hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/util/StreamerUtil.java

```diff
@@ -314,7 +314,7 @@ public static boolean partitionExists(String tablePath, String partitionPath, or
    * Generates the bucket ID using format {partition path}_{fileID}.
    */
   public static String generateBucketKey(String partitionPath, String fileId) {
-    return String.format("%s_%s", partitionPath, fileId);
+    return partitionPath + "_" + fileId;
```

Comment:
If you want the best micro performance strictly, `StringBuilder` should be used.

Reply:
yeah, I found this doc: https://dip-mazumder.medium.com/stringbuilder-vs-string-in-java-a-guide-for-optimal-memory-usage-4a284d8243ea#:~:text=In%20conclusion%2C%20when%20it%20comes,can%20lead%20to%20performance%20inefficiencies.

Reply:
Also I found this: https://stackoverflow.com/questions/1532461/stringbuilder-vs-string-concatenation-in-tostring-in-java, the Java compiler will automically optimize the concantenation into stringBuilder actually.

Reply:
But Java compiler unnecessary introduces overhead, so if we really care about this micro performance, `StringBuilder` must be used.